src/test/index.ts

```diff
@@ -31,8 +32,89 @@ type TestFixtures = PlaywrightTestArgs & PlaywrightTestOptions & {
 type WorkerAndFileFixtures = PlaywrightWorkerArgs & PlaywrightWorkerOptions & {
   _browserType: BrowserType;
   _artifactsDir: () => string,
+  _reuseBrowerContext: ReuseBrowerContextStorage,
 };
 
+class ReuseBrowerContextStorage {
+  private _browserContext?: BrowserContext;
+  private _uniqueOrigins = new Set<string>();
+  private _options?: BrowserContextOptions;
+  private _pauseNavigationEventCollection = false;
+
+  isEnabled(): boolean {
+    return !!process.env.PWTEST_REUSE_CONTEXT;
+  }
+
+  async obtainContext(browser: Browser, newContextOptions: BrowserContextOptions): Promise<BrowserContext> {
+    if (!this._browserContext)
+      return await this._createNewContext(browser);
+    return await this._refurbishExistingContext(newContextOptions);
+  }
+
+  private async _createNewContext(browser: Browser): Promise<BrowserContext> {
+    this._browserContext = await browser.newContext();
+    this._options = (this._browserContext as any)._options;
+    this._browserContext.on('page', page => page.on('framenavigated', frame => {
+      if (this._pauseNavigationEventCollection)
+        return;
+      this._uniqueOrigins.add(new URL(frame.url()).origin);
+    }));
+    return this._browserContext;
+  }
+
+  async _refurbishExistingContext(newContextOptions: BrowserContextOptions): Promise<BrowserContext> {
+    assert(this._browserContext);
+    const pages = this._browserContext.pages();
+    const page = pages[0];
+    this._pauseNavigationEventCollection = true;
+    const initialOrigin = new URL(page.url()).origin;
+    while (this._uniqueOrigins.size > 0) {
+      const nextOrigin = this._uniqueOrigins.has(initialOrigin) ? initialOrigin : this._uniqueOrigins.values().next().value;
+      this._uniqueOrigins.delete(nextOrigin);
+      await page.route(nextOrigin, route => route.fulfill({ body: `<html></html>`, contentType: 'text/html' }));
```

Comment:
Can we route origins? I would install one route and point all documents to empty html

Reply:
What do you mean with that? temporary **/* is probably the best if this is what you mean. No need to manually filter for origins there.